### PR TITLE
commit rule testing

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
Current dependencies on/for this PR:
* feature/pos-fast-hotstuff-consensus
  * **PR #742** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/742" target="_blank">Graphite</a> 
    * **PR #758** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/758" target="_blank">Graphite</a> 
      * **PR #775** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/775" target="_blank">Graphite</a> 
        * **PR #776** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/776" target="_blank">Graphite</a>  👈
      * **PR #761** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/761" target="_blank">Graphite</a> 
        * **PR #762** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/762" target="_blank">Graphite</a> 
          * **PR #774** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/774" target="_blank">Graphite</a> 
            * **PR #759** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/759" target="_blank">Graphite</a> 
              * **PR #765** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/765" target="_blank">Graphite</a> 
                * **PR #766** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/766" target="_blank">Graphite</a> 
                  * **PR #771** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/771" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/776?utm_source=stack-comment).

File: lib/pos_blockchain_test.go [hunk 1]
```diff
@@ -1428,12 +1444,257 @@ func TestCanCommitGrandparent(t *testing.T) {
...
+	blockTemplate.Header.ProposerVotePartialSignature = dummySig
+	return blockTemplate
 }
 
 func _generateRandomBLSPrivateKey(t *testing.T) *bls.PrivateKey {
 	privateKey, err := bls.NewPrivateKey()
 	require.NoError(t, err)
 	return privateKey
 }
+
```
Comment (tholonious) [inline lib/pos_blockchain_test.go:1640]: Add a brief comment here to describe the what this test verifies, and the list of cases.

File: lib/pos_blockchain_test.go [hunk 2]
```diff
@@ -1428,12 +1444,257 @@ func TestCanCommitGrandparent(t *testing.T) {
 	// TODO: What other cases do we really need tested here?
 }
 
-func TestCommitGrandparent(t *testing.T) {
-	t.Skip("Skipping TestCommitGrandparent")
+func TestRunCommitRuleOnBestChain(t *testing.T) {
```
Comment (tholonious) [inline lib/pos_blockchain_test.go:1447]: It'll be good add a comment here to describe the what this test verifies, and the list of cases. This way, it's easier for future readers to eyeball the test.

File: lib/pos_blockchain.go [hunk 3]
```diff
@@ -841,7 +828,7 @@ func (bc *Blockchain) commitBlock(blockHash *BlockHash) error {
 	// Connect a view up to the parent of the block we are committing.
 	utxoView, err := bc.getUtxoViewAtBlockHash(*block.Header.PrevBlockHash)
 	if err != nil {
-		return errors.Wrapf(err, "runCommitRuleOnBestChain: Problem initializing UtxoView: ")
+		return errors.Wrapf(err, "commitBlock: Problem initializing UtxoView: ")
```
Comment (tholonious) [inline lib/pos_blockchain.go:831]: Should this function be called `commitBlockPoS` to match the naming pattern of other PoS blockchain functions?

**Review by tholonious — APPROVED** [review]

File: lib/pos_blockchain.go (hunk 3 above)
Comment (lazynina) [reply lib/pos_blockchain.go:831]: Let's address this rename later in the stack

File: lib/pos_blockchain_test.go (hunk 1 above)
Comment (lazynina) [reply lib/pos_blockchain_test.go:1640]: This is a test helper function, but I'll add a comment about how it does some setup for PoS blockchain testing.

File: lib/pos_blockchain.go (hunk 3 above)
Comment (lazynina) [reply lib/pos_blockchain.go:831]: addressed in #790

**Pull request merged**